### PR TITLE
Send jobseeker alert data to google sheets

### PR DESCRIPTION
**Pull request by pezholio**
## Trello card URL:

https://trello.com/c/U2LgZDU1/822-send-jobseeker-alert-data-to-google-sheets-so-that-it-is-easier-to-measure-feature-performance

## Changes in this PR:

This adds a new `AuditData` record every time a user creates a subscription, with the reference used, as well as the search criteria. As with other `AuditData` records, there's a Rake task that gathers all the data from that day and adds them to the spreadsheet.

To Do

- [x] Add a Rake task to backfill the subscriptions we already have
- [ ] Add a new sheet to the stats spreadsheet (dev and prod)
- [ ] Add the GID of the new sheet to the tfvars file(s)

## Next steps:

- [x] Terraform deployment required?

- [x] New development configuration to be shared?

### CODE REVIEW

**Comment by pezholio** [comment]
[Send jobseeker alert data to Google Sheets so that it is easier to measure feature performance](https://trello.com/c/U2LgZDU1/822-send-jobseeker-alert-data-to-google-sheets-so-that-it-is-easier-to-measure-feature-performance)

File: variables.tf [hunk 1]
```diff
@@ -364,7 +369,7 @@ variable "google_analytics_profile_id" {}
 variable "audit_vacancies_worksheet_gid" {}
 variable "audit_feedback_worksheet_gid" {}
 variable "audit_express_interest_worksheet_gid" {}
-
+variable "audit_subscription_creation_gid" {}
```
Comment (erbridge) [inline variables.tf]: Tiny nit pick, but having the new line here helps with readability.

File: variables.tf [hunk 2]
```diff
@@ -158,6 +158,11 @@ variable "send_job_alerts_daily_email_command" {
   default     = ["rake", "verbose", "daily_emails:send"]
 }
 
+variable "backfill_jobseeker_alert_data_command" {
+    description = "The Entrypoint for the backfill_jobseeker_alert_data task"
+  default     = ["rake", "verbose", "jobseeker_alerts:statistics:backfill:alert_data"]
```
Comment (erbridge) [inline variables.tf]: Nit pick: whitespace is a mess here.

File: spec/lib/tasks/update_spreadsheets_rake_spec.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
+require 'rails_helper'
+
+RSpec.describe 'rake spreadsheets:update', type: :task do
```
Comment (erbridge) [inline spec/lib/tasks/update_spreadsheets_rake_spec.rb]: :+1:

File: spec/controllers/subscriptions_controller_spec.rb [hunk 4]
```diff
@@ -26,25 +30,39 @@
...
+              email: '<script>foo@email.com</script>',
+              search_criteria: "<body onload=alert('test1')>Text</body>",
+              frequency: "<img src='http://url.to.file.which/not.exist' onerror=alert(document.cookie);>"
+            }
+          }
+        end
 
-        subscription = Subscription.last
-        expect(subscription).to be_nil
+        it 'does not allow unsafe parameters' do
```
Comment (erbridge) [inline spec/controllers/subscriptions_controller_spec.rb]: Redundancy in this test name, now.

File: docker-compose.env.sample [hunk 5]
```diff
@@ -19,3 +19,4 @@ NOTIFY_SUBSCRIPTION_CONFIRMATION_TEMPLATE=
 SUBSCRIPTION_KEY_GENERATOR_SECRET=10dc5b4d246466a71aeb9fd87a41207b6ec77a38096cf3f5a0948981f5c873a9f7cd1c3444cfc7cdc8e2b37dbabfc31e0c35a597f552e683eb471555e8c75834
 SUBSCRIPTION_KEY_GENERATOR_SALT=fc811c17d10be2aa5467237eef8cb8c8f6edb493fcc120c1b9e2db67159cce073398d660167370c24d5eadab56ee93ba15f72e3824ada843e43232ec31640393
 FEATURE_EMAIL_ALERTS=true
+AUDIT_SUBSCRIPTION_CREATION_GID=
```
Comment (erbridge) [inline docker-compose.env.sample]: Small request: can we put this next to the spreadsheet ID, so it's easier to parse what this refers to?

**Pull request merged**